### PR TITLE
Bsd port os perf bsd per cpu fbsd

### PR DESCRIPTION
**Pull request by devnexen**
To cherry pick.

### CODE REVIEW

File: src/hotspot/os/bsd/os_perf_bsd.cpp [hunk 1]
```diff
@@ -120,8 +120,17 @@ size_t length = sizeof(cpu_load_info);
     if (sysctl(mib, miblen, &cpu_load_info, &length, NULL, 0) == -1) {
       return OS_ERR;
     }
+#elif defined(__FreeBSD__)
+    size_t alllength = length * nProcs * sizeof(long);
+    long *allcpus = NEW_C_HEAP_ARRAY(long, alllength, mtInternal);
```
Comment (bsdkurt) [inline src/hotspot/os/bsd/os_perf_bsd.cpp]: The NEW_C_HEAP_ARRAY macro looks like this:
```
#define NEW_C_HEAP_ARRAY(type, size, memflags)\
  (type*) (AllocateHeap((size) * sizeof(type), memflags))
```
so the size passed to it is not in bytes but in terms of type. e.g. `nProcs * CPUSTATES` is correct here.

File: src/hotspot/os/bsd/os_perf_bsd.cpp [hunk 2]
```diff
@@ -120,8 +120,17 @@ size_t length = sizeof(cpu_load_info);
...
+
+    if (sysctlbyname("kern.cp_times", allcpus, &alllength, NULL, 0) == -1) {
+      return OS_ERR;
```
Comment (bsdkurt) [inline src/hotspot/os/bsd/os_perf_bsd.cpp:129]: `sysctlbyname` third parameter is in bytes but `alllength` is not correctly set. length is already in bytes so multiplying to by sizeof(long) is not correct. `size_t alllength = length * nProcs;` is correct for this value.

Also the memory allocated for allcpus is leaked. Please use `FREE_C_HEAP_ARRAY` to deallocate.

**Review by bsdkurt — CHANGES_REQUESTED** [review]
Thanks for working on this. Please see comments below.

File: src/hotspot/os/bsd/os_perf_bsd.cpp (hunk 1 above)
Comment (devnexen) [reply src/hotspot/os/bsd/os_perf_bsd.cpp]: Ah right I missed the sizeof bit...

**Comment by bsdkurt** [comment]
Looks good to me now. Thanks.

**Comment by bsdkurt** [comment]
BTW, if you make a pull request against bsdkurt:bsd-port-os-perf-bsd I can merge it in and it will become apart of #39.

**Comment by bsdkurt** [comment]
Changes from this pull request have been merged into #39. Closing

**Pull request closed without merging**